File: selfstudy/python/words_sortasc.py
'''
task 199:
Words - SortAsc
You need to sort the values in ascending order for word lengths
'''

def sort_up_words(lengths):
    return sorted(lengths)

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], sort_up_words([10, 20, 30])),
        ([50, 60, 70], sort_up_words([50, 60, 70])),
        ([1, 2, 3], sort_up_words([1, 2, 3])),
        ([100, 200], sort_up_words([100, 200])),
        ([], sort_up_words([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = sort_up_words(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# works i think

File: selfstudy/js/words_sortasc.js
/*
task 99:
Words - SortAsc
You need to sort the values in ascending order for word lengths
*/

function sort_up_words(lengths) {
  return [...lengths].sort((a, b) => a - b);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: sort_up_words([10, 20, 30]) },
  { input: [50, 60, 70], expected: sort_up_words([50, 60, 70]) },
  { input: [1, 2, 3], expected: sort_up_words([1, 2, 3]) },
  { input: [100, 200], expected: sort_up_words([100, 200]) },
  { input: [], expected: sort_up_words([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('sort_up_words', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(sort_up_words(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, sort_up_words(t.input) === t.expected || JSON.stringify(sort_up_words(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// works i think
